File: frontend/pages/projects.py
import streamlit as st
from src.api.client import APIClient

if "token" not in st.session_state or not st.session_state.token:
    st.warning("Please log in on the Home page.")
    st.stop()

api = APIClient(token=st.session_state.token)

st.title("📂 Project Management")

# Fetch projects using the client
res = api.get_projects()
if res.status_code == 200:
    projects = res.json()
    for p in projects:
        with st.expander(p['title']):
            st.write(p['description']) # type: ignore
            st.caption(f"Created: {p['created_at']}")

File: frontend/src/api/projects.ts
import { apiClient } from "./client";

export interface Project {
    id: number;
    title: string;
    description: string | null;
    created_at: string;
    last_modified: string;
};

export const projectsService = {
    getAll: async () => {
        const response = await apiClient.get<Project[]>('/projects');
        return response.data;
    },
    create: async (title: string, description: string = '') => {
        const response = await apiClient.post<Project>('/projects/create', {
            title,
            description: description,
        });
        return response.data;
    }
};